### DISCUSSION TITLE
M1 wrong -mach-o, but wrong architecture

### DISCUSSION THREAD

**Discussion by L1uNan** [fastapi/fastapi#6646] [Questions]
Im getting an error in my terminal when trying to run. in the MacBook m1 virtual python environment .

heres my trace:

(school-backend-NxQ28yof-py3.9) LN@Mac  ~/Documents/sensorsPro/miniprogram_backend   develop ±  uvicorn miniprogram_backend.main:app --reload INFO: Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit) INFO: Started reloader process [24366] using statreload Process SpawnProcess-1: Traceback (most recent call last): File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap self.run() File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 108, in run self._target(*self._args, **self._kwargs) File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/subprocess.py", line 61, in subprocess_started target(sockets=sockets) File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/server.py", line 49, in run loop.run_until_complete(self.serve(sockets=sockets)) File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete return future.result() File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/server.py", line 56, in serve config.load() File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/config.py", line 308, in load self.loaded_app = import_from_string(self.app) File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/importer.py", line 23, in import_from_string raise exc from None File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/uvicorn/importer.py", line 20, in import_from_string module = importlib.import_module(module_str) File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "<frozen importlib._bootstrap>", line 1030, in _gcd_import File "<frozen importlib._bootstrap>", line 1007, in _find_and_load File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 680, in _load_unlocked File "<frozen importlib._bootstrap_external>", line 855, in exec_module File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed File "/Users/L/Documents/sensorsPro/miniprogram_backend/./miniprogram_backend/main.py", line 4, in <module> from fastapi import FastAPI File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/fastapi/__init__.py", line 7, in <module> from .applications import FastAPI as FastAPI File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/fastapi/applications.py", line 3, in <module> from fastapi import routing File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/fastapi/routing.py", line 18, in <module> from fastapi import params File "/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/fastapi/params.py", line 4, in <module> from pydantic.fields import FieldInfo ImportError: dlopen(/Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/pydantic/__init__.cpython-39-darwin.so, 2): no suitable image found. Did find: /Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/pydantic/__init__.cpython-39-darwin.so: mach-o, but wrong architecture /Users/L/Library/Caches/pypoetry/virtualenvs/school-backend-NxQ28yof-py3.9/lib/python3.9/site-packages/pydantic/__init__.cpython-39-darwin.so: mach-o, but wrong architecture

i don't know what to do...
could you give me some advice please?

**Comment by ShimronAlakkal** [comment]
I got a similar error too. I think it's something related to the architecture of the M1 machines. I thinks fastapi doesn't have a support for the M1 architecture as of now. I think we've to add a feature request then.

By the way if you've found a workaround or a solution, that would help me a lot.

Thanks

**Comment by JustKievn** [comment]
In case anyone else stumbles across this with the same issue. Running terminal with Rosetta and reinstalling packages solved this issue for me.

**Comment by baebae-dev** [comment]
I had a similar problem trying to install dask. I solved it following these steps:

1) Check which version of python are installed (I have : mac os python 2.x and 3.x, anaconda 3.8, brew 3.9):
```
python -m pipenv.help
Python installations found:
```
3.9.6: /opt/homebrew/bin/python3
3.9.6: /opt/homebrew/bin/python3.9
3.8.8: /opt/anaconda3/bin/python3
3.8.8: /opt/anaconda3/bin/python3.8
3.8.2: /usr/bin/python3
2.7.16: /usr/bin/python2
2.7.16: /usr/bin/python2.7

2) Use pipenv to create a 3.8 virtual environment:
```
pipenv install --python '/opt/anaconda3/bin/python3'
pipenv install 'dask[complete]'
```
Note: by default, pipenv was using the 3.9 brew version of python to create the virtual environment - which caused the problem.

3) Activate the virtual environment and use it:
```
pipenv shell
```
